### PR TITLE
Fix: support to set layout direction for hybrid composition platform view

### PR DESCRIPTION
**Pull request by wangying3426**
On android platform, virtual display platform view can set direction layout correctly, but hybrid composition will fail and throw an NullPointerException as https://github.com/flutter/flutter/issues/96003 shown, this PR fixes it.

Besides, if a virtual display platform view is not found with specified viewId,  `vdControllers.get(viewId)` will be null, so an NPE will also occur, this PR also fixes it.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by wangying3426** [comment]
> It looks like this pull request may not have tests. Please make sure to add tests before merging. If you need [an exemption](https://github.com/flutter/flutter/wiki/Tree-hygiene#tests) to this rule, contact Hixie on the #hackers channel in [Chat](https://github.com/flutter/flutter/wiki/Chat).
> 
> If you are not sure if you need tests, consider this rule of thumb: the purpose of a test is to make sure someone doesn't accidentally revert the fix. Ask yourself, is there anything in your PR that you feel it is important we not accidentally revert back to how it was before your fix?
> 
> **Reviewers**: Read the [Tree Hygiene page](https://github.com/flutter/flutter/wiki/Tree-hygiene#how-to-review-code) and make sure this patch meets those guidelines before LGTMing.

@Hixie Could you please review this RP? thanks.

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java [hunk 1]
```diff
@@ -122,6 +122,7 @@
   private final PlatformViewsChannel.PlatformViewsHandler channelHandler =
       new PlatformViewsChannel.PlatformViewsHandler() {
 
+        @TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
```
Comment (blasten) [inline shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java:125]:
```suggestion
        @TargetApi(Build.VERSION_CODES.KITKAT)
```

Comment (blasten) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java:125]: this is expected below, see comment.

File: shell/platform/android/test/io/flutter/plugin/platform/PlatformViewsControllerTest.java [hunk 2]
```diff
@@ -385,6 +385,39 @@ public void createPlatformViewMessage__throwsIfViewHasParent() {
...
+    // Simulate create call from the framework.
+    createPlatformView(jni, platformViewsController, platformViewId, "testType", /* hybrid=*/ true);
+    assertEquals(ShadowFlutterJNI.getResponses().size(), 1);
+
+    // Simulate set direction call from the framework.
+    setLayoutDirection(jni, platformViewsController, platformViewId, 1);
+    // The limit value of reply message will be equal to 2 if the layout direction is set
+    // successfully, otherwise it will be much more than 2 due to the reply message contains
+    // an error message wrapped with exception detail information.
+    assertEquals(ShadowFlutterJNI.getResponses().get(0).limit(), 2);
```
Comment (blasten) [inline shell/platform/android/test/io/flutter/plugin/platform/PlatformViewsControllerTest.java:418]: what about also adding `verify(androidView, times(1)).setLayoutDirection(1);`?

File: shell/platform/android/test/io/flutter/plugin/platform/PlatformViewsControllerTest.java [hunk 3]
```diff
@@ -385,6 +385,39 @@ public void createPlatformViewMessage__throwsIfViewHasParent() {
...
+    platformViewsController.getRegistry().registerViewFactory("testType", viewFactory);
+
+    FlutterJNI jni = new FlutterJNI();
+    attach(jni, platformViewsController);
+
+    // Simulate create call from the framework.
+    createPlatformView(jni, platformViewsController, platformViewId, "testType", /* hybrid=*/ true);
+    assertEquals(ShadowFlutterJNI.getResponses().size(), 1);
+
+    // Simulate set direction call from the framework.
```
Comment (blasten) [inline shell/platform/android/test/io/flutter/plugin/platform/PlatformViewsControllerTest.java:413]: what about also adding verify(androidView, never()).setLayoutDirection(anyInt());  to ensure that it wasn't called prior to the next line?

**Comment by wangying3426** [comment]
@blasten @chinmaygarde Could you have any other suggestions？

**Review by blasten — APPROVED** [review]
lgtm

**Pull request merged**